pkg/scheduler/cache/util.go

```diff
@@ -58,3 +62,157 @@ func createShadowPodGroup(pod *v1.Pod) *v1alpha1.PodGroup {
 		},
 	}
 }
+
+func GenerateJob(mpi metav1.Object, job batchv1.Job) *batchv1.Job {
+	// TODO: Default and Validate these with Webhooks
```

Comment:
should we build those job by a new operator?

Reply:
I don't _think_ we need a separate operator to handle this. It feels cleaner keeping all the MPI logic in a single place. I could be swayed otherwise.